File: backend/app/schemas/note.py
from pydantic import BaseModel


class NoteCreate(BaseModel):
    CustomerID: int | None = None
    LeadID: int | None = None
    DealID: int | None = None
    NoteText: str
    CreatedBy: int


class NoteUpdate(BaseModel):
    CustomerID: int | None = None
    LeadID: int | None = None
    DealID: int | None = None
    NoteText: str
    CreatedBy: int

File: frontend/types/note.ts
export interface Note {
    NoteID: number;
    CustomerID?: number | null;
    LeadID?: number | null;
    DealID?: number | null;
    NoteText: string;
    CreatedBy: number;
  }
  
  export interface NoteCreate {
    CustomerID?: number | null;
    LeadID?: number | null;
    DealID?: number | null;
    NoteText: string;
    CreatedBy: number;
  }
  
  export interface NoteUpdate {
    CustomerID?: number | null;
    LeadID?: number | null;
    DealID?: number | null;
    NoteText: string;
    CreatedBy: number;
  }